libparsec/crates/client/src/workspace/transactions/open_file.rs

```diff
@@ -50,6 +50,88 @@ impl OpenOptions {
         }
     }
 }
+#[derive(Debug, thiserror::Error)]
+pub enum WorkspaceIsFileContentLocalError {
+    #[error("Component has stopped")]
+    Stopped,
+    #[error(transparent)]
+    Internal(#[from] anyhow::Error),
+
+    // resolve_path related errors
+    #[error("Cannot communicate with the server: {0}")]
+    Offline(#[from] ConnectionError),
+    #[error("Not allowed to access this realm")]
+    NoRealmAccess,
+    #[error("Path doesn't exist")]
+    EntryNotFound,
+    #[error(transparent)]
+    InvalidKeysBundle(#[from] Box<InvalidKeysBundleError>),
+    #[error(transparent)]
+    InvalidCertificate(#[from] Box<InvalidCertificateError>),
+    #[error(transparent)]
+    InvalidManifest(#[from] Box<InvalidManifestError>),
+
+    // specific errors
+    #[error("Path resolves to a folder, not a file")]
+    NotAFile,
+}
+
+/// To check if all blocks are locally present
+/// to avoid unexpected errors when trying to
+/// open offline a file with missing blocks
```

Comment:
A couple of suggestion based on rustdoc book on [Documenting components](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components):

```suggestion
/// Checks if all blocks are locally present for the given file `path`
///
/// This check allows to avoid unexpected errors when trying to open a file with 
/// missing blocks while being offline.
///
/// No check is performed if `path` is a folder, instead an error is returned.
```